pkg/app/controller.go

```diff
@@ -127,34 +130,28 @@ func (f *HostForwarder) Handler() HTTPHandler {
 		if host == "" {
 			return fmt.Errorf("invalid url missing host value: %q", r.URL.String())
 		}
-		hostURL, err := f.URLResolver.GetHostURL(zone, host)
+		proxy, err := f.buildProxy(zone, host)
 		if err != nil {
 			return err
 		}
-		proxy := newHostReverseProxy(host, hostURL)
 		proxy.ServeHTTP(w, r)
 		return nil
 	}
 }
 
-type hostReverseProxy struct {
-	innerProxy *httputil.ReverseProxy
-}
-
-func newHostReverseProxy(hostName string, hostBaseURL *url.URL) *hostReverseProxy {
-	director := func(r *http.Request) {
-		r.URL.Scheme = hostBaseURL.Scheme
-		r.URL.Host = hostBaseURL.Host
-		split := strings.SplitN(r.URL.Path, "hosts/"+hostName, 2)
-		r.URL.Path = split[1]
+func (f *HostForwarder) buildProxy(zone, host string) (*httputil.ReverseProxy, error) {
```

Comment:
nit: buildReverseProxy, keeping the reverse term.

Reply:
Done